### PR TITLE
Enhancement of the MinIO admin client functionality with the group function

### PR DESCRIPTION
**Pull request by Vedat-D**
Three functions have been added to the client.

A function (**addUpdateGroup**) to add a group with optional group membership or to update the membership of a Group. 
A function (**removeGroup**) to delete a group.
**Example**: 
Create an empty group: 
`minioClient.addUpdateGroup(groupName, Status.ENABLED, null);`
Update membership of group :
`minioClient.addUpdateGroup(groupId, Status.ENABLED, groupMembers);`
Delete a group: 
`minioClient.removeGroup(groupName);`

A function (**getGroupInfo**) to get information of a group.
**Example**: 
Get info of a group:
 `GroupInfo groupInfo = minioClient.getGroupInfo(groupName);`

A function (**listGroups**) to list all groups.
**Example**: 
List all groups: 
`List<String> groupsMinio = minioClient.listGroups();`

I have extracted the enum status but left it in the class UserInfo so as not to cause any breaking changes.
However, it is recommended to replace the code occurances "UserInfo.Status" with "Status" and then delete the code duplicate of UserInfo

### CODE REVIEW

**Review by balamurugana — CHANGES_REQUESTED** [review]

File: adminapi/src/main/java/io/minio/admin/GroupAddUpdateRemoveInfo.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage,
+ * (C) 2021 MinIO, Inc.
```
Comment (harshavardhana) [inline adminapi/src/main/java/io/minio/admin/GroupAddUpdateRemoveInfo.java:3]:
```suggestion
 * (C) 2022 MinIO, Inc.
```

File: adminapi/src/main/java/io/minio/admin/GroupInfo.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage,
+ * (C) 2021 MinIO, Inc.
```
Comment (harshavardhana) [inline adminapi/src/main/java/io/minio/admin/GroupInfo.java:3]:
```suggestion
 * (C) 2022 MinIO, Inc.
```

File: adminapi/src/main/java/io/minio/admin/Status.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package io.minio.admin;
```
Comment (harshavardhana) [inline adminapi/src/main/java/io/minio/admin/Status.java:1]: missing copyright header..

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by balamurugana — APPROVED** [review]

**Comment by Vedat-D** [comment]
Hello together, what is the status here?

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**